replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/ReplicatorRunner.java

```diff
@@ -74,23 +84,33 @@ public ReplicatorRunner(
   ReplicatorRunner(
       ScheduledExecutorService scheduledExecutor,
       Replicator replicator,
+      Heartbeater heartbeater,
       ReplicatorConfigManager replicatorConfigManager,
       SiteManager siteManager,
       Security security) {
     this.scheduledExecutor = notNull(scheduledExecutor);
     this.replicator = notNull(replicator);
+    this.heartbeater = notNull(heartbeater);
     this.replicatorConfigManager = notNull(replicatorConfigManager);
     this.siteManager = notNull(siteManager);
     this.security = security;
   }
 
   public void init() {
-    long period =
+    final long replicationPeriod =
         Long.parseLong(
             System.getProperty("org.codice.replication.period", DEFAULT_REPLICATION_PERIOD_STR));
+    final long heartbeatPeriod =
```

Comment:
Do we need a separate executor/thread to schedule the heartbeat separate from the normal replication?

Reply:
When Peter and I discussed this last week, we came up with that to avoid blocking replication while we were pinging and awaiting response from sites. This simplified a bit the design.